client/solve.go

```diff
@@ -181,10 +183,7 @@ func (c *Client) solve(ctx context.Context, def *llb.Definition, runGateway runG
 		defer cancelSolve()
 
 		defer func() { // make sure the Status ends cleanly on build errors
-			go func() {
-				<-time.After(3 * time.Second)
```

Comment:
Couldn't figure out why this 3 second delay is here but I'm thinking it's safe to remove if we are handling context cancellation later? Happy to add it back if there was a reason for it.

Reply:
When solve call closes, the server closes the status as well. This delay is in here to give a fallback if for whatever reason(faulty daemon, network issues) status remains stuck and does not close.

Reply:
> When solve call closes, the server closes the status as well. This delay is in here to give a fallback if for whatever reason(faulty daemon, network issues) status remains stuck and does not close.

Ah, thanks for the explanation. Are you able to point me to the code where this happens?